pyro/distributions/multivariate_normal.py

```diff
@@ -29,15 +29,30 @@ def linear_solve_compat(matrix, matrix_chol, y):
 
 def matrix_inverse_compat(matrix, matrix_chol):
     """Computes the inverse of a positive semidefinite square matrix."""
-    assert matrix.requires_grad == matrix_chol.requires_grad
-    if matrix.requires_grad:
+    if matrix.requires_grad or matrix_chol.requires_grad:
```

Comment:
Could you help me understand under what conditions the assertion would fail?

Reply:
The problem here is potrf_compat on PyTorch < 0.3.0: In that case `Variable(torch.potrf(sigma.data))` is returned and if `sigma.requires_grad` was true then they don't match up any more. The unit tests didn't catch this until now because they did not need any gradients.